src/fsharp/IlxGen.fs

```diff
@@ -5860,10 +6212,17 @@ and GenBindingAfterDebugPoint cenv cgbuf eenv sp (TBind(vspec, rhsExpr, _)) star
         | Local (_, realloc, _), Expr.Const (Const.Zero, _, _) when not realloc && not (eenv.isInLoop && vspec.IsMutable) ->
             CommitStartScope cgbuf startScopeMarkOpt
         | _ ->
+            GetStoreValCtxt cenv cgbuf eenv vspec
             GenBindingRhs cenv cgbuf eenv SPSuppress vspec rhsExpr
             CommitStartScope cgbuf startScopeMarkOpt
             GenStoreVal cenv cgbuf eenv vspec.Range vspec
 
+and GetStoreValCtxt cenv cgbuf eenv (vspec: Val) =
+    // Emit the ldarg0 if needed
+    match StorageForVal cenv.g vspec.Range vspec eenv with
+    | Env (ilCloTy, _, _) -> CG.EmitInstr cgbuf (pop 0) (Push [ilCloTy]) mkLdarg0
```

Comment:
This feels like it's specifically related to closures due to the name `ilCloTy` - is that the case?

Reply:
Does not look like it, I see the code was lifted here.